tests/everest/test_everserver.py

```diff
@@ -271,3 +274,52 @@ def test_everserver_status_max_batch_num(
         filter_out_gradient=False, batches=None
     )
     assert {data.batch for data in snapshot.simulation_data} == {0}
+
+
+@patch("sys.argv", ["name", "--config-file", "config_minimal.yml"])
+@patch("everest.detached.jobs.everserver._configure_loggers")
+@patch("everest.detached.jobs.everserver._generate_authentication")
+@patch(
+    "everest.detached.jobs.everserver._generate_certificate",
+    return_value=(None, None, None),
+)
+@patch(
+    "everest.detached.jobs.everserver._find_open_port",
+    return_value=42,
+)
+@patch("everest.detached.jobs.everserver._write_hostfile")
+@patch("everest.detached.jobs.everserver._everserver_thread")
```

Comment:
Looks like a lot of duplication in this test file, so might consider making a fixture to avoid said duplication, something like:

```python
@pytest.fixture()
def mock_server(monkeypatch):
    monkeypatch.setattr(everest.detatched.jobs.everserver, "_configure_loggers", MagicMock())
    ...
```
or

```python
@pytest.fixture()
def mock_server(monkeypatch):
    with patch(..), patch(...) ...
        yield
```

Reply:
I think it is a good idea. I will move the duplicate patching to a fixture.